include/caffe/net.hpp

```diff
@@ -255,6 +258,8 @@ class Net {
   size_t memory_used_;
   /// Whether to compute and display debug info for the net.
   bool debug_info_;
+  /// Auto top blob index to be incremented in GetAutoTopBlobName
+  size_t current_auto_top_blob_index_;
```

Comment:
I think this needs to be explicitly initialized or its value will technically be undefined?